server/src/api/taskratings/taskratings.controller.ts

```diff
@@ -10,23 +10,74 @@ import Customer from '../customer/customer.model';
 
 export const getTaskratings: RouteHandlerFnc = async (ctx) => {
   const { user, role } = ctx.state;
-  if (!user || !role) {
-    throw new Error('User and role are required');
-  }
+  if (!user || !role) throw new Error('User and role are required');
 
   const query = Taskrating.query()
     .where({ parentTask: Number(ctx.params.taskId) })
     .modify('keepVisible', user, role);
-  if (ctx.query.eager) {
-    query.withGraphFetched('[createdBy, createdFor]');
-  }
+  if (ctx.query.eager) query.withGraphFetched('[createdBy, createdFor]');
   ctx.body = await query;
 };
 
-export const postTasksRatings: RouteHandlerFnc = async (ctx) => {
-  if (!ctx.state.user) {
-    throw new Error('User is required');
-  }
+export const postTaskRatings: RouteHandlerFnc = async (ctx) => {
+  if (!ctx.state.user) throw new Error('User is required');
+  const ratings = ctx.request.body;
+  const userId = Number(ctx.state.user.id);
+  const roadmapId = Number(ctx.params.roadmapId);
+  const taskId = Number(ctx.params.taskId);
+
+  if (!Array.isArray(ratings)) return void (ctx.status = 400);
+
+  const hasValueRatePermission = hasPermission(ctx, Permission.TaskValueRate);
+  const hasWorkRatePermission = hasPermission(ctx, Permission.TaskWorkRate);
+
+  const errors = await Promise.all(
+    ratings.map(async (rating) => {
+      if (
+        (rating.dimension === TaskRatingDimension.BusinessValue &&
+          !hasValueRatePermission) ||
+        (rating.dimension === TaskRatingDimension.RequiredWork &&
+          !hasWorkRatePermission)
+      )
+        return { err: 403 };
+      if (rating.forCustomer !== undefined) {
+        // check that user is representative for given customer
+        const found = await Customer.relatedQuery('representatives')
+          .for(rating.forCustomer)
+          .findById(userId);
+        if (!found) return { err: 403 };
+      }
+      return {};
+    }),
+  );
+
+  if (errors.some(({ err }: any) => err === 403))
+    return void (ctx.status = 403);
+
+  const newRatings = await Taskrating.transaction(
+    async (trx) =>
+      await Promise.all(
+        ratings.map(
+          async ({ dimension, value, comment, forCustomer }: any) =>
+            await Task.relatedQuery('ratings', trx)
```

Comment:
```suggestion
   (trx) =>
     Promise.all(
        ratings.map(
         ({ dimension, value, comment, forCustomer }: any) =>
           Task.relatedQuery('ratings', trx)
```

The `async () => await ...` can be simplified to `() => ...` which returns equivalent promise.